simulation/agents.py

```diff
@@ -479,6 +480,8 @@ def update_heating_status(self, model: "DomesticHeatingABM") -> None:
         proba_failure = probability_density * step_interval_years
         if random.random() < proba_failure:
             self.heating_functioning = False
+        else:
+            self.heating_functioning = True
```

Comment:
WDYT about replacing this with `self.heating_functioning = not self.true_with_probability(proba_failure)` or `self.true_with_probability(1 - proba_failure)` ?

Reply:
I feel like the overall step flow (the fact that it gets reset back to functioning) is easier to read in current version